deploy/build_install_deploy.sh

```diff
@@ -54,23 +70,29 @@ INSTALL_DIR="/home/$USER/chronolog/$BUILD_TYPE"
 BIN_DIR="${INSTALL_DIR}/bin"
 CONF_DIR="${INSTALL_DIR}/conf"
 
-# Print a message with a green background indicating that the build process is starting
-echo -e "\033[42;30mBuilding ChronoLog in ${BUILD_TYPE} mode...\033[0m"
-"${SCRIPT_DIR}/build.sh" -type "$BUILD_TYPE"
+# Check if the required scripts exist and are executable
+for script in build.sh install.sh local_single_user_deploy.sh; do
+    if [[ ! -x "${SCRIPT_DIR}/${script}" ]]; then
+        echo -e "${RED}Error: ${script} not found or not executable in ${SCRIPT_DIR}${RESET}"
+        exit 1
+    fi
+done
 
-# Print a message with a green background indicating that the install process is starting
-echo -e "\033[42;30mInstalling ChronoLog...\033[0m"
-"${SCRIPT_DIR}/install.sh"
+# Build ChronoLog
+echo -e "${GREEN}Building ChronoLog in ${BUILD_TYPE} mode...${RESET}"
+"${SCRIPT_DIR}/build.sh" -type "$BUILD_TYPE"
 
-# Print a message with a green background indicating that the deployment process is starting
-echo -e "\033[42;30mDeploying ChronoLog...\033[0m"
+# Install ChronoLog
+echo -e "${GREEN}Installing ChronoLog...${RESET}"
+"${SCRIPT_DIR}/install.sh" -type "$BUILD_TYPE"
 
-# Run the deployment script with the specified parameters and standard directories
+# Deploy ChronoLog
+echo -e "${GREEN}Deploying ChronoLog...${RESET}"
 "${SCRIPT_DIR}/local_single_user_deploy.sh" \
     -n "$NUM_KEEPERS" \
     -j "$NUM_GRAPHERS" \
     -w "$INSTALL_DIR" \
     ${DEPLOY:+-d} ${RESET:+-r} ${STOP:+-s} ${KILL:+-k}
```

Comment:
the parameter expansion used here does not check the value of those operation variables. as long as they are set and no empty, the `-d`, `-r`, `-s` and `-k` will be appended to the command.